### PR TITLE
Bugfix/stay in context callback

### PR DESCRIPTION
**Pull request by JanTrotnow**
The current state's instance was missing as a `stayInContext`/`clearContext` callback parameter in state machine. I just fixed that so that you can access the state instance in your callback implementation.

### CODE REVIEW

**Comment by antoniusostermann** [comment]
@JanTrotnow Hier fehlt noch immer die Abarbeitung der CodeClimate-Issues. Bitte morgen erledigen. Diejenigen Issues, die nicht zutreffen oder nicht behebbar sind, bitte entsprechend in CodeClimate markieren ("invalid" bzw. "wontfix") und entsprechend begründen.

**Comment by JanTrotnow** [comment]
I just finished fixing my codeclimate issues. Can't resolve the remaining two.

File: src/components/state-machine/public-interfaces.ts [hunk 1]
```diff
@@ -174,3 +174,20 @@ export interface ContextState {
...
+export type StayInContextCallback = (
+  currentStateName?: string,
+  currentStateInstance?: State.Required,
+  contextStateNames?: string[],
+  intentHistory?: Array<{ stateName: string; intentMethodName: string }>,
+  stateNameToTransitionTo?: string
+) => boolean;
+
+/** Callback that is executed to determine wether or not the state context gets cleared */
+export type ClearContextCallback = (
```
Comment (antoniusostermann) [inline src/components/state-machine/public-interfaces.ts]: Would be awesome if you could add some jsdocs explaining all parameters.

In addition, you got a typo ("wether").

File: src/components/state-machine/state-machine.ts [hunk 2]
```diff
@@ -110,8 +113,9 @@ export class StateMachine implements Transitionable {
 
     /* Execute callbacks of context states and filter by result */
     contextStates = contextStates.filter(contextState =>
-      (this.retrieveStayInContextCallbackFromMetadata(contextState.instance.constructor as State.Constructor) as ((...args: any[]) => boolean))(
+      (this.retrieveStayInContextCallbackFromMetadata(contextState.instance.constructor as State.Constructor) as StayInContextCallback)(
```
Comment (antoniusostermann) [inline src/components/state-machine/state-machine.ts]: It would be better to accept a generic in your method and pass it here instead of explicit casting

**Pull request merged**